### ISSUE TITLE
Undefined variable: dummy

### ISSUE THREAD

**Issue by WhyNotHugo** [lukas-reineke/indent-blankline.nvim#11]
Installed this (alongside with `indentLine`, which I was already using), but I get this error on startup (and no indentation guide on blank lines):

    indent_blankline encountered an error: Vim(let):E121: Undefined variable: dummy

**Comment by lukas-reineke** [comment]
`dummy` is not a variable from this plugin.
It sounds like something is not initializing properly in the embedded Neovim instance with your setup.

Can you search your config for `dummy`?
It might also be another plugin, you can try disabling them.

**Comment by WhyNotHugo** [comment]
- Disabled `indent_blankline`: error goes away.
- Enable `indent_blankline`: error comes back.

Searching my plugins for the work `dummy` yields 413 results, so it's not really feasible to track down usage elsewhere.

Any other hints on how to debug this?

**Comment by lukas-reineke** [comment]
i tried to install all your plugins, but i still can't reproduce the issue. Does it happen for a specific file type?

to debug, you can try to comment out the catch block in `autoload/indent_blankline/embedded.vim`, line 35 and 36

restart nvim and call it directly with `:call indent_blankline#embedded#FindMatches(expand('%'), 0)`

if you get `Error invoking 'vim_eval' on channel 1:^@Invalid channel: 1`, then its working as it should
i think this is where the variable error happens, it might have more information

**Comment by WhyNotHugo** [comment]
It used to happen on python files. Now I'm getting no error output, but indentation lines are not drawn on empty lines (eg: plugin isn't working at all).

Running the above command:
```
Error detected while processing function indent_blankline#embedded#FindMatches:
line   32:
Error invoking 'vim_eval' on channel 1:^@Invalid channel: 1
Press ENTER or type command to continue
```

Running the above command after commenting out that `catch` block:

```
Error detected while processing function indent_blankline#embedded#FindMatches[19]..GetPythonIndent:
line   63:
E121: Undefined variable: dummy
Press ENTER or type command to continue
```

It happens mostly on Python files. I don't think it's happened on other file types.

**Comment by lukas-reineke** [comment]
i released an update to improve the error logging, and also turn it off by default

you can set `g:indent_blankline_debug = v:true` to get the logs back


I found the issue now. `GetPythonIndent` uses a very questionable "trick" to break out of a loop. It just crashes.

https://github.com/vim/vim/blob/cb711abf0f71d8c743cf73c76077f52e17732a8c/runtime/indent/python.vim#L73-L79

The way Vim handles exceptions is super weird. I didn't fully understand until now.
Basically, a function will normally continue to run after an error, but if it is called inside of an try/catch block, it will exit early.
So the "trick" didn't work when this plugin calls it, it just aborted the function.
To not exit the function early, I have to run it with `silent!`

tldr: i fixed it now

Thanks for the issue and debug help, i would have never found this on my own



The default Python indent expression is still not perfect, there are a bunch of cases where empty lines get no correct level. But at least it should not break everything now.
I use https://github.com/Vimjas/vim-python-pep8-indent which gives much better results, but thats just FYI

**Comment by WhyNotHugo** [comment]
>  Basically, a function will normally continue to run after an error, but if it is called inside of an try/catch block, it will exit early.

Wow, that's very unusual. I'd never have guessed that.

Thanks for addressing this, works fine for me now!

> I use https://github.com/Vimjas/vim-python-pep8-indent which gives much better results, but thats just FYI

I'll have a look, I do think it always gets that wrong. :)
